### PR TITLE
Fixes #163 - Add pre-ingest test for matching subjects.

### LINKED ISSUE #163: Subject import tool - add pre-ingest test for matching subjects
### Descriptive summary

Add a new Pre-ingest validation function to test for matching subject strings before the import is run.

### Rationale

Currently we don't know which subjects in a new ingest are already in the DAMS so we reconcile all of them and then run the subject import.  Commonly, many of the subjects are already in the DAMS and are ignored by the import tool.  A considerable amount of time could be saved in the preparation of the ingest spreadsheet if these subjects were identified and eliminated.

### Expected behavior

Add a new Pre-ingest validation function "Check for exsiting subjects" to the subjectImport.do webform
Return a .csv report with the following columns:
 - subject type
 - subject term
 - result ("match", "no match")
 - ARK
 - exactMatch
 - closeMatch

If multiple matches are found (topic, genre, name), place each on a new row.
Email DAMS Manager Invocation Result to user.

### PR DESCRIPTION
**Pull request by lsitu**
Fixes 163

Add pre-ingest test for matching subjects to the Subject Import tool.

@ucsdlib/developers - please review

### CODE REVIEW

File: WebContent/WEB-INF/jsp/subjectImport.jsp [hunk 1]
```diff
@@ -112,6 +112,10 @@
                             <input checked type="radio" name="preingestOption" value="pre-processing" checked>
                             <span class="submenuText">Preview the converted RDF/XML only, no ingest.</span>
                         </div>
+                        <div title="Check this checkbox to pre-ingest test for matching subjects." class="submenuText">
```
Comment (mcritchlow) [inline WebContent/WEB-INF/jsp/subjectImport.jsp]: I wonder if we should consider re-wording the title to something like: `Select this option test for matching subjects prior to ingest` or something? As it is, there might be a little ambiguity about exactly will happen.

File: src/edu/ucsd/library/xdre/tab/SubjectMatching.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+			try {
+				boolean result = false;
+				Document srcDoc = saxReader.read(rdfFile);
+				Node labelNode = srcDoc.selectSingleNode("//mads:authoritativeLabel");
+				String subjectTerm  = labelNode.getText();
+				String modelName= getSubjectTypeLabel(labelNode.getParent().getName());
+				
+				for (String model : models) {
+					List<String> matchedSubjects = damsRepository.findAuthority(model, subjectTerm);
+					if (matchedSubjects != null) {
```
Comment (mcritchlow) [inline src/edu/ucsd/library/xdre/tab/SubjectMatching.java]: could we flatten the conditional check here into a guard clause? something like:
```
if (matchedSubjects == null) {
  continue;
}
...
```

File: WebContent/WEB-INF/jsp/subjectImport.jsp (hunk 1 above)
Comment (lsitu) [reply WebContent/WEB-INF/jsp/subjectImport.jsp]: Sound good.

File: src/edu/ucsd/library/xdre/tab/SubjectMatching.java (hunk 2 above)
Comment (lsitu) [reply src/edu/ucsd/library/xdre/tab/SubjectMatching.java]: Yes. we can.

**Comment by lsitu** [comment]
Done! See commit https://github.com/ucsdlib/damsmanager/pull/199/commits/2f3831516e1ca29174db8cbc4fe68edb79822bf2.

**Pull request merged**